### PR TITLE
notifications: general marketing

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-934

## What changed (plus any additional context for devs)
added simple handling for marketing focused notifications, we already technically had support for them but now it allows us to navigate to different screens

## Screen recordings / screenshots
https://cloud.skylarbarrera.com/RPReplay_Final1701273411.MP4

## What to test

android, can't build locally atm 😮‍💨

### CODE REVIEW

**Comment by walmat** [comment]
Taking a look after expo meeting

**Review by ibrahimtaveras00 — APPROVED** [review]
Looked good on Android as well, QA Passed ✅ 
https://github.com/rainbow-me/rainbow/assets/41752922/f44b8ff7-55db-46cc-9035-1bdd495518a3

also did a check on notifications from WC and mainnet txns 👍🏽

File: src/notifications/NotificationsHandler.tsx [hunk 1]
```diff
@@ -300,6 +301,19 @@ export const NotificationsHandler = ({ walletReady }: Props) => {
         `NotificationsHandler: handling wallet connect notification`,
         { notification }
       );
+    } else if (type === NotificationTypes.marketing) {
+      logger.info(`NotificationsHandler: handling marketing notification`, {
+        notification,
+      });
+      const data = (notification.data as unknown) as MarketingNotificationData;
```
Comment (benisgold) [inline src/notifications/NotificationsHandler.tsx:308]: idk what double type casting does but just noting it in case its not necessary

Comment (skylarbarrera) [reply src/notifications/NotificationsHandler.tsx:308]: can't cast it to `MarketingNotificationData` without casting to unknown first, TS bs

**Pull request merged**